src/arch/runtime/context_switching.cc

```diff
@@ -292,7 +289,7 @@ artificial_stack_t::artificial_stack_t(void (*initial_fun)(void), size_t _stack_
 #elif defined(__s390x__)
     sp -= 16; // r6-r13 and f8-f15.
 #elif defined (__powerpc64__)
-    sp += 24;
+    sp -= 20; // r14-r31, toc, cr.
```

Comment:
Yeah, it's safe to do on x86 because none of the SSE/floating point registers are callee-save (and so aren't 'live' on entry to the swapcontext function). On POWER and s390x however the ABIs have callee-save floating point registers and the compiler assumes they will be restored after a function call.

In practice the compiler probably doesn't use the callee-saved floating point registers very often in code like this, so not saving/restoring them might not cause any issues. But it is a bit fragile IMO. Worth a comment anyway.